test/parser_test.exs

```diff
@@ -774,189 +751,226 @@ defmodule Surface.Compiler.ParserTest do
+                  ], %{has_sub_blocks?: true, line: 1, file: "nofile", column: 3}}
                ]
     end
 
     test "handle invalid parents for #else" do
       code = """
       <div>
-      <#else>
+      {#else}
       </div>
       """
 
       exception = assert_raise ParseError, fn -> parse!(code) end
 
-      message = "cannot use <#else> inside <div>. Possible parents are \"<#if>\" and \"<#for>\""
+      message =
+        "no valid parent node defined for {#else}. Possible parents are \"{#if}\" and \"{#for}\""
+
       assert %ParseError{message: ^message, line: 2} = exception
     end
 
     test "handle invalid parents for #elseif" do
       code = """
       <div>
-      <#elseif>
+      {#elseif}
       </div>
       """
 
       exception = assert_raise ParseError, fn -> parse!(code) end
 
       message =
-        "cannot use <#elseif> inside <div>. The <#elseif> construct can only be used inside a \"<#if>\""
+        "no valid parent node defined for {#elseif}. The {#elseif} construct can only be used inside a \"{#if}\""
 
       assert %ParseError{message: ^message, line: 2} = exception
     end
 
     test "handle invalid parents for #match" do
       code = """
       <div>
-      <#match>
+      {#match}
       </div>
       """
 
       exception = assert_raise ParseError, fn -> parse!(code) end
 
       message =
-        "cannot use <#match> inside <div>. The <#match> construct can only be used inside a \"<#case>\""
+        "no valid parent node defined for {#match}. The {#match} construct can only be used inside a \"{#case}\""
 
       assert %ParseError{message: ^message, line: 2} = exception
     end
 
     test "raise error on sub-blocks without parent node" do
       code = """
         1
-      <#else>
+      {#else}
         2
       """
 
       exception = assert_raise ParseError, fn -> parse!(code) end
 
       message =
-        "no valid parent node defined for <#else>. Possible parents are \"<#if>\" and \"<#for>\""
+        "no valid parent node defined for {#else}. Possible parents are \"{#if}\" and \"{#for}\""
 
       assert %ParseError{message: ^message, line: 2} = exception
     end
+
+    test "raise error on unknown block open" do
+      code = """
+      {#iff}
+        1
+      {/if}
+      """
+
+      exception = assert_raise ParseError, fn -> parse!(code) end
+
+      message = """
+      unknown `{#iff}` block. Available blocks are \
+      "if", "unless", "for", "case", "else", "elseif" and "match"\
+      """
+
+      assert %ParseError{message: ^message, line: 1} = exception
+    end
+
+    test "raise error on unknown block close" do
+      code = """
+      {#if}
+        1
+      {/iff}
+      """
+
+      exception = assert_raise ParseError, fn -> parse!(code) end
+
+      message = """
+      unknown `{/iff}` block. Available blocks are \
+      "if", "unless", "for", "case", "else", "elseif" and "match"\
```

Comment:
Propose `case`, `else`, `elseif` and `match` may mislead the user as these constructs cannot be directly closed with their equivalent: `{/case}`, `{/else}`, `{/elseif}` and `{/match}` I suggest to remove them.